profiles/services/combined.py

```diff
@@ -22,8 +22,14 @@
 #### ID is required - no getting around it ####
 
 
-def get_by_id(sso_email_id: str) -> Profile:
+def get_by_id(sso_email_id: str, include_inactive: bool = False) -> Profile:
+    """
+    Retrieve a profile by its User ID.
+    """
     # NB we're not raising more specific exceptions here because we're optimising this for speed
+    if include_inactive:
+        return Profile.objects.get(sso_email_id=sso_email_id)
+
     return Profile.objects.get(sso_email_id=sso_email_id, is_active=True)
```

Comment:
```suggestion
    is_active = include_inactive or None
    return Profile.objects.get(sso_email_id=sso_email_id, is_active=is_active)
```
Not sure if this is actually works or not, but worth considering? Obviously for all the similar functions in here...